<?xml version="1.0" encoding="UTF-8" ?>
<launch>
    <include file="$(find rr_iarrc)/launch/drag_centerline_planner.launch"/>
    <include file="$(find rr_iarrc)/launch/laplacian_line_detector.launch"/>
    <node pkg="rr_platform" type="image_transform" name="image_transform" output="screen">
        <param name="transform_topics" value="/camera_center/image_color_rect/lines/detection_img"/>
        <param name="px_per_meter" value="50"/>
        <param name="map_dist_max" value="3.5"/>
        <param name="map_dist_min" value="0.0"/>

        <param name="fallback_fov_horizontal" value="1.6955"/>
        <param name="fallback_fov_vertical" value="1.3758"/>
        <param name="fallback_image_width" value="1280"/>
        <param name="fallback_image_height" value="964"/>
    </node>


    <include file="$(find rr_iarrc)/launch/startlight_watcher.launch"/>
    <include file="$(find rr_iarrc)/launch/color_detector.launch"/>
    <include file="$(find rr_iarrc)/launch/race_end_detector.launch"/>

    <node name="navigation_controller" pkg="rr_common" type="navigation_controller" output="screen">
        <param name="req_finish_line_crosses" type="int" value="1"/>
        <param name="startSignal" type="string" value="/start_detected"/>
        <param name="resetSignal" type="string" value="/reset_detected"/>
        <param name="finishLineCrossesSignal" type="string" value="/finish_line_crosses"/>
    </node>
</launch>
